src/query/procedure/module.cpp

```diff
@@ -133,6 +135,20 @@ void RegisterMgLoad(ModuleRegistry *module_registry, utils::RWLock *lock, Builti
   module->AddProcedure("load", std::move(load));
 }
 
+namespace {
+[[nodiscard]] bool IsFileEditable(const std::optional<std::filesystem::path> &path) {
+  return path && access(path->c_str(), W_OK) == 0;
```

Comment:
Ah, that is true, I didn't take that into account. Good catch!

Reply:
What do you think should be displayed for path then? It should contain builtin and shared libs.
"unknown", "compiled"?

Reply:
I think the path is fine as a real path, because it helps the users: I would like to change the X module, where it is located? . I was thinking about introducing just a check in the `IsFileEditable` function: `of !path.ends_with(".py") return false;` (pseudo code). But to be honest I am not sure about this, as from a query procedure point of view, there is not difference between python and compiled query modules.

Reply:
We return to the user `is_editable`. We can communicate to the user that that represents if the **module** is editable, not just the file. That would require that the file is editable + that the language is interpreted.

Reply:
But technically the compiled files are also editable with the query modules, as nothing will stop a query module to modify a shared lib. Of course it would be super cumbersome to somehow communicate the content of the module, but technically it is possible (e.g. it is possible to write a query procedure that uses base64 format, then there is no problem to overwrite the binary modules). I am not saying this something we (or anybody else) should do, just to point out that we have to clearly communicate what `is_editable` mean.

Reply:
Maybe the original implementation is enough (it returns the filesystem based permissions), and the filtering for `.py` files will be done in the query module. I think this is the cleanest way I can think of.

Reply:
Hmm leaving it to the edit procedures is a good idea. We just provide the necessary information and leave the more complex logic to the user.

Reply:
Procedures should take care of that you cannot edit shared libs, only the python files. 

I will then leave the solution to the current one, return all the paths and if their editable, no matter what type they are.